forms/HtmlEditorField.php

```diff
@@ -578,7 +578,7 @@ public function getanchors() {
 
 			// Similar to the regex found in HtmlEditorField.js / getAnchors method.
 			if (preg_match_all("/\s(name|id)=\"([^\"]+?)\"|\s(name|id)='([^']+?)'/im", $page->Content, $matches)) {
```

Comment:
This regex should really get cleaned up - it's not massively clear what the 2nd position even is in this case.